include/ublksrv.h

```diff
@@ -648,6 +648,22 @@ extern const struct ublksrv_dev *ublksrv_dev_init(const struct ublksrv_ctrl_dev
  */
 extern void ublksrv_dev_deinit(const struct ublksrv_dev *dev);
 
+/**
+ * Parses the standard command line options used when adding a device
+ * and populates the ublksrv_dev_data structure
+ *
+ * @param data the ublksrv device to populate from the command line arguments.
+ * @param efd whether the command line arguments specified an eventfd to be used
+ * @param argc number of arguments
+ * @param argv argument list
+ */
+extern int ublksrv_parse_std_opts(struct ublksrv_dev_data *data, int *efd, int argc, char *argv[]);
```

Comment:
IMO, command line parsing should belong to application instead of library, and different application is supposed to take different command line handling.

Reply:
I agree in general but I think these arguments are not application specific but are all about how the ublksrv userspace runtime will interact with the kernel component, so probably all ublk targets should use these options.

Then in addition to these standard options a specific target can have a second loop of parsing command line arguments process the target-specific arguments.
Like how ublk.null/loop/nbd already do. They all take the device standard options and then they have a second loop for the special options.
I don't see why an external target should not want to provide these standard options.

But I do not have a strong opinion either for or against. If you do not think this belongs in the public api then I am ok with that and you can just close this PR.

Reply:
Or maybe even the whole of ublksrv_cmd_dev_add() should be part of the public api.